Adamant/Modules/Settings/Notifications/NotificationSounds/NotificationSoundsFactory.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  NotificationSoundsFactory.swift
+//  Adamant
+//
+//  Created by Yana Silosieva on 20.08.2024.
+//  Copyright © 2024 Adamant. All rights reserved.
+//
+
+import Swinject
+import SwiftUI
+
+struct NotificationSoundsFactory {
+    private let parent: Assembler
+    private let assemblies = [NotificationSoundAssembly()]
+    
+    init(parent: Assembler) {
+        self.parent = parent
+    }
+    
+    @MainActor
+    func makeView(target: NotificationTarget) -> NotificationSoundsView {
+        let assembler = Assembler(assemblies, parent: parent)
+        let viewModel = {
+            assembler.resolver.resolve(NotificationSoundsViewModel.self)!
+        }()
+        
+        viewModel.setup(notificationTarget: target)
```

Comment:
Let's move the viewModel out of the DI to pass this parameter through init.

And I suggest to build the assembler inside the escaping as well. I know that in other places it's not inside the escaping, but it's more productive to keep it inside.